src/key.h

```diff
@@ -128,6 +128,18 @@ class CKey
      */
     bool SignCompact(const uint256& hash, std::vector<unsigned char>& vchSig) const;
 
+    /**
+     * Create a BIP-340 Schnorr signature, for the xonly-pubkey corresponding to *this,
+     * optionally tweaked by *merkle_root. Additional nonce entropy can be provided through
+     * aux.
+     *
+     * When merkle_root is not nullptr, this results in a signature with a modified key as
```

Comment:
What's the use case for non-tweaked signatures?

nit: The difference between `merkle_root == nullptr` and `merkel_root->IsNull()` is subtle enough that
one might be mistaken, especially when passing the `aux` parameter and not reading the comments closely. Could this be a potential problem in the future?

Reply:
Signatures with OP_CHECKSIG and friends aren't tweated; only key-path signatures are.

So we do need both a way to ask for untweaked, and for tweaked with nothing. I agree it's confusing, though we need a way to ask for both. Getting it wrong is exceedingly likely to cause validation failures, though.

Reply:
Yikes, I missed that subtlety too. In #22275 you expand the comment bit, e.g.:
```
* When merkle_root is nullptr, as distinct from merkle_root->IsNull(), the signature is not tweaked.
* This is the case for e.g. OP_CHECKSIG in a tapscript. For key-path signature on the other hand the signature is always tweaked, as follows: 
```

Reply:
Added something like that to https://github.com/bitcoin/bitcoin/pull/22275.